src/zepben/evolve/streaming/get/network_consumer.py

```diff
@@ -130,103 +152,139 @@ async def y():
 
     async def _get_network_hierarchy(self) -> GrpcResult[NetworkHierarchy]:
         return await self.try_rpc(self._handle_network_hierarchy)
 
     async def _get_feeder(self, service: NetworkService, mrid: str) -> GrpcResult[MultiObjectResult]:
-        feeder_response = await self._get_identified_object(service, mrid)
-        if feeder_response.was_failure:
-            return feeder_response
+        feeder_response = (await self._get_identified_object(service, mrid)).throw_on_error()
         feeder: Feeder = feeder_response.result
 
         if not feeder:
             return GrpcResult(result=None)
 
-        equipment_objects = await self._get_identified_objects(service, service.get_unresolved_reference_mrids(resolver.ec_equipment(feeder)))
-        if equipment_objects.was_failure:
-            return equipment_objects
-
-        resolvers = list()
-        resolvers.append(resolver.normal_energizing_substation(feeder))
-        for equip in feeder.equipment:
-            try:
-                for terminal in equip.terminals:
-                    resolvers.append(resolver.connectivity_node(terminal))
-                if isinstance(equip, Conductor):
-                    resolvers.append(resolver.per_length_sequence_impedance(equip))
-                    resolvers.append(resolver.asset_info(equip))
-            except AttributeError:
-                pass  # Not ConductingEquipment.
-            resolvers.append(resolver.psr_location(equip))
-
-        mrids = service.get_unresolved_reference_mrids(resolvers)
-        objects = await self._get_identified_objects(service, mrids)
-        if objects.was_failure:
-            return objects
-        objects.result.value.update(equipment_objects.result.value)  # Combine with previous results
-        objects.result.value[feeder.mrid] = feeder  # Add feeder to result
-        return GrpcResult(result=MultiObjectResult(objects.result.value, objects.result.failed.union(equipment_objects.result.failed)))
+        if not isinstance(feeder, Feeder):
+            return GrpcResult(result=ValueError(f"Requested mrid {mrid} was not a Feeder, was {type(feeder)}"))
+
+        mor = MultiObjectResult()
+        res = (await self._get_equipment_for_container(service, feeder.mrid)).throw_on_error()
+        mor.value.update(res.result.value)
+        # We need to resolve all the unresolved references for each piece of equipment, but then we need to
+        # also do the same for the resolutions, and so on so forth until the entire tree of references originating from the Feeder has been resolved,
+        # however we don't know how long this tree is until we've resolved everything. So we start by resolving all the equipment, then we iteratively
+        # descend the tree by checking for unresolved references for the objects we just resolved in the previous iteration.
+        to_resolve = self._get_unresolved_mrids(service, res.result.value.keys(), _EXCLUDED_GET_FEEDER)
```

Comment:
Why are other feeders being excluded? This is information that should be brought back and with the removal of equipment ID's is not going to cause the creep it used to

Reply:
do we want the hierarchy to come back with get_feeder though? i was thinking the typical usage pattern would be to get the network hierarchy and then from that get a feeder

Reply:
@kgreav I'm a bit confused why we would pull the whole hierarchy back to get the feeder when the specific feeder has been requested?

Reply:
no, i was saying that would be a typical usage pattern for a user and that we wouldn't want get_feeder to return/fetch the network hierarchy.

anyway this is all moot now, feeders wont be excluded, but you'll only get other feeders if some equipment belongs to them (but you won't get those feeders equipment).

Reply:
@gcarwardine The entire hierachy will be pulled in if there are any open points on the requested feeder that have a different feeder on the other side of the switch due to those other feeders being added to the recursive resolve loop

@kgreav Given that it is highly likely that the feeder has open points with at least one other feeder you can porbably just include the selected feeder in the recursive loop rather than having a second lookup for the substation only

Reply:
yep, good point, i've moved it into the loop.